pkg/controller/kogitoapp/resource/image_metadata.go

```diff
@@ -0,0 +1,141 @@
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/openshift"
+
+	appsv1 "github.com/openshift/api/apps/v1"
+	dockerv10 "github.com/openshift/api/image/docker10"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	prometheusLabelKeyPrefix = "prometheus.io"
+	labelNamespaceSep        = "/"
+	orgKieNamespaceLabelKey  = "org.kie" + labelNamespaceSep
+	dockerLabelServicesSep   = ","
+	portSep                  = ":"
+	defaultExportedProtocol  = "http"
+	portFormatWrongMessage   = "Service %s on " + openshift.ImageLabelForExposeServices + " label in wrong format. Won't be possible to expose Services for this application. Should be PORT_NUMBER:PROTOCOL. e.g. 8080:http"
+)
+
+func dockerImageHasLabels(dockerImage *dockerv10.DockerImage) bool {
+	if dockerImage == nil || dockerImage.Config == nil || dockerImage.Config.Labels == nil {
+		return false
+	}
+	return true
+}
+
+// mergeImageMetadataWithDeploymentConfig retrieves org.kie and prometheus.io labels from DockerImage and adds them to the DeploymentConfig
+// returns true if any changes occurred in the deploymentConfig based on the dockerImage labels
+func mergeImageMetadataWithDeploymentConfig(dc *appsv1.DeploymentConfig, dockerImage *dockerv10.DockerImage) bool {
+	if !dockerImageHasLabels(dockerImage) {
+		return false
+	}
+
+	log.Debugf("Preparing to read docker labels and add them to the Deployment: %s", dockerImage.Config.Labels)
+
+	if dc.Spec.Template.Annotations == nil {
+		dc.Spec.Template.Annotations = map[string]string{}
+	}
+	if dc.Labels == nil {
+		dc.Labels = map[string]string{}
+	}
+	if dc.Spec.Selector == nil {
+		dc.Spec.Selector = map[string]string{}
+	}
+	if dc.Spec.Template.Labels == nil {
+		dc.Spec.Template.Labels = map[string]string{}
+	}
+
+	added := false
+	for key, value := range dockerImage.Config.Labels {
+		if strings.Contains(key, orgKieNamespaceLabelKey) {
+			splitedKey := strings.Split(key, labelNamespaceSep)
+			// we're only interested on keys like org.kie/something
+			if len(splitedKey) == 2 {
+				importedKey := splitedKey[len(splitedKey)-1]
+
+				if !added {
+					_, lblPresent := dc.Labels[importedKey]
+					_, selectorPresent := dc.Spec.Selector[importedKey]
+					_, podLblPresent := dc.Spec.Template.Labels[importedKey]
+
+					added = !(lblPresent && selectorPresent && podLblPresent)
+				}
+
+				dc.Labels[importedKey] = value
+				dc.Spec.Selector[importedKey] = value
+				dc.Spec.Template.Labels[importedKey] = value
+			}
+		} else if strings.Contains(key, prometheusLabelKeyPrefix) {
+			if !added {
+				_, present := dc.Spec.Template.Annotations[key]
+				added = !present
+			}
+
+			dc.Spec.Template.Annotations[key] = value
+		}
+	}
+
+	return added
+}
+
+// discoverPortsAndProbesFromImage set Ports and Probes based on labels set on the DockerImage of this DeploymentConfig
+func discoverPortsAndProbesFromImage(dc *appsv1.DeploymentConfig, dockerImage *dockerv10.DockerImage) {
```

Comment:
Can you please add test coverage for this method, making sure that exposed service label was correctly processed and correct probes were created?

Reply:
I'll take a look, but I guess we already have. This function is old, we just reallocated in this file.

Reply:
ok, I just didn't find it in image_metadata_test.go

Reply:
I'll add a specific one